src/name_theme.ts

```diff
@@ -99,25 +101,72 @@ const TAROT = [
   "The World",
 ]
 
+// https://github.com/wu-qing-157/Anonymous/blob/3b3f087942ba6bf209cda689b65cce98ab79d61b/app/src/main/java/org/wkfg/anonymous/utils.kt#L300
+class randomN {
+  // we have to use bigint to handle the random algorithm
+  a: JSBI
+  b: JSBI = JSBI.BigInt(19260817) // +1s
+
+  constructor(seed: number) {
+    this.a = JSBI.BigInt(seed)
+  }
+
+  truncate = (t: JSBI) => {
+    return JSBI.asIntN(64, t)
+  }
+
+  next = () => {
+    let t = this.a
+    let s = this.b
+    this.a = s
+    t = this.truncate(JSBI.bitwiseXor(t, JSBI.leftShift(t, JSBI.BigInt(23))))
+    t = this.truncate(
+      JSBI.bitwiseXor(t, JSBI.signedRightShift(t, JSBI.BigInt(17)))
+    )
+    t = this.truncate(
+      JSBI.bitwiseXor(
+        JSBI.bitwiseXor(t, s),
+        JSBI.signedRightShift(s, JSBI.BigInt(26))
+      )
+    )
+    this.b = t
+
+    // 9223372036854775807 is the Long.MAX_VALUE in Kotlin
+    return JSBI.bitwiseAnd(JSBI.add(s, t), JSBI.BigInt("9223372036854775807"))
+  }
+}
+
+// https://github.com/wu-qing-157/Anonymous/blob/3b3f087942ba6bf209cda689b65cce98ab79d61b/app/src/main/java/org/wkfg/anonymous/utils.kt#L318
+function shuffle(nameList: string[], seed: number) {
```

Comment:
The only thing I'm wondering is performance. For each floor (and expand reply floor), we need to shuffle once, which yields `O(Floor * N)` complexity, plus constant factor introduced by `JSBI` calculation. What about using a global hash map to cache `(scheme, seed) -> shuffle result`?

Reply:
Fair point. Fixed in #52 .